### DISCUSSION TITLE
Trouble upgrading ozone redistribution code for GC3.1 MM

### DISCUSSION THREAD

**Discussion by lcj10** [MetOffice/simulation-systems#238] [UM]
Hi

I'm trying to get some code set up with HadGEM3-GC31-MM with ozone redistribution. I've tried various other things, but am currently trying to upgrade the ozone code in a copy of the 1% run from CMIP6 (u-bg618) using the instructions here https://code.metoffice.gov.uk/trac/moci/wiki/OzoneRedistribution Something is going wrong and I'm getting an error on the command line with verification (see below). Does anyone know what this is referring to and what I can do? Thanks

Laura

[INFO] REGISTERED u-di961 -> /home/h01/hadlr/cylc-run/u-di961
[FAIL] cylc validate -o /var/tmp/tmpM3VtyV --strict u-di961 # return-code=1, stderr=
[FAIL] "'-1': not a valid cylc-shorthand or full ISO 8601 date representation"

**Comment by james-bruten-mo** [comment]
Hi Laura, are you using Cylc 8 for this and if so do you know whether that suite has been used with Cylc8 before? I think the ISO8601 error may be a Cylc7->Cylc8 difference in which case you'll need to edit that part of the suite. The Workflows Guild on yammer may be a better place to get some help if this is the case.

Another thing to try is adding `--debug` to the `cylc validate` command which may give some useful extra output.

**Reply by lcj10** [reply]
Thanks James. No I'm using cylc7. I can't work out where the cylc validate command actually comes from - doing a grep of the suite doesn't give anything useful.

**Comment by ericaneininger** [comment]
The error `[FAIL] "'-1': not a valid cylc-shorthand or full ISO 8601 date representation"` is the result of this suite trying to run with a cycle period of 0 days (RESUB=0).  All offsets, for cycling purposes, are calculated from this jinja2 variable including the shorthand date representation in the graph - which in turn Cylc is unable to verify.  You cannot run with a zero length cycle period. I'm happy to admit that the error message is not particularly helpful here.
